hudi-common/src/main/java/org/apache/hudi/common/data/HoodieData.java

```diff
@@ -196,4 +212,42 @@ default <O> HoodieData<T> distinctWithKey(SerializableFunction<T, O> keyGetter,
         .reduceByKey((value1, value2) -> value1, parallelism)
         .values();
   }
+
+  /**
+   * The key used in a caching map to identify a {@link HoodieData}.
+   *
+   * At the end of a write operation, we manually unpersist the {@link HoodieData} associated with that writer.
+   * Therefore, in multi-writer scenario, we need to use both {@code basePath} and {@code instantTime} to identify {@link HoodieData}s.
+   */
+  class HoodieDataCacheKey implements Serializable {
```

Comment:
We should avoid exposing this outside of the `HoodieData` class (no other components should be exposed to how we're caching it, so it would be easier for us to change if we need to)